packages/template-retail-react-app/app/hooks/use-update-shopper-context.js

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import {useEffect} from 'react'
+import {useLocation} from 'react-router-dom'
+import {
+    useUsid,
+    useShopperContext,
+    useShopperContextsMutation
+} from '@salesforce/commerce-sdk-react'
+import {isServer, isHydrated} from '@salesforce/retail-react-app/app/utils/utils'
+import {useQueryClient} from '@tanstack/react-query'
+
+// Hooks
+import useMultiSite from '@salesforce/retail-react-app/app/hooks/use-multi-site'
+
+// Constants
+import {useShopperContextSearchParams} from '@salesforce/retail-react-app/app/hooks/use-shopper-context-search-params'
+
+/*
+ * This hook will set the shopper context when search params pertinant
+ * to shopper context are present.
+ */
+export const useUpdateShopperContext = () => {
+    const {site} = useMultiSite()
+    const {usid} = useUsid()
+    const {search} = useLocation()
+    const queryClient = useQueryClient()
+    const createShopperContext = useShopperContextsMutation('createShopperContext')
+    const updateShopperContext = useShopperContextsMutation('updateShopperContext')
+    const {data: shopperContext} = useShopperContext(
+        {parameters: {usid, siteId: site.id}},
+        {enabled: !isServer}
+    )
```

Comment:
I saw that there are _multiple_ requests sent to query the shopper context, and I'm not sure why exactly. 

We should look into this because of performance. If this proves to be tricky, I'm ok with revisiting this in another PR.

Reply:
This could be intentional. The first time we get the context, the sequential time is either updating context or client cache is cleared after hydration

Reply:
Yes so I tested this and here are my findings:

1. No shopper context + No query params
    - GET /shopper context API call x1
    <img width="1783" alt="Screenshot 2024-09-17 at 9 53 26 AM" src="https://github.com/user-attachments/assets/ec1b9d9a-1c11-4f09-a2d2-71b9f5b2a1e5">
2. No shopper context + query params
    - GET /shopper context API calls x2
    <img width="1496" alt="Screenshot 2024-09-17 at 10 12 09 AM" src="https://github.com/user-attachments/assets/a3a4de24-90b5-4887-a2a0-752fe1029906">
3. shopper context exists + no query params
    - GET /shopper context API calls x2 on refresh
    <img width="1486" alt="Screenshot 2024-09-17 at 10 18 29 AM" src="https://github.com/user-attachments/assets/eb3d5118-2c31-42db-9677-cb86d78dccf8">
4. shopper context exists + query params
    - GET /shopper context API calls x3
    <img width="1365" alt="Screenshot 2024-09-17 at 10 17 16 AM" src="https://github.com/user-attachments/assets/c2a98312-57b2-4836-bc69-c93048fb2340">

Reply:
I guess the 2 that make more GETs than necessary are 
- 3 (GET when shopper context exists) 
- 4 (PATCH shopper-context)

Reply:
I've updated the code to only refetch the data if there are no pending shopper context updates. This way it only makes 2 GET calls during a PATCH.
1. First when the page loads
2. when the update completes